mbed_host_tests/host_tests_plugins/host_test_plugins.py

```diff
@@ -129,6 +130,31 @@ def check_mount_point_ready(self, destination_disk, init_delay=0.2, loop_delay=0
                 self.print_plugin_char('.')
         return result
 
+    def check_serial_port_ready(self, serial_port, target_id=None):
+        result = True
+
+        if target_id:
+            mbeds = mbed_lstools.create()
+            mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
+
+            # Wait for mount point to appear with mbed-ls
+            # and if it does check if mount point for target_id changed
+            # If mount point changed, use new mount point and check if its ready (os.access)
+            new_serial_port = serial_port
+            for i in range(25): # 25x 200ms = 5sec
+                if target_id in mbeds_by_tid:
+                    if 'serial_port' in mbeds_by_tid[target_id]:
+                        new_serial_port = mbeds_by_tid[target_id]['serial_port']
+                        break
+                sleep(200)
```

Comment:
```mbeds = mbed_lstools.create()``` should be done inside the loop. Otherwise it will loop on same data.
Like:
```
            for i in range(25): # 25x 200ms = 5sec
                mbeds = mbed_lstools.create()
                mbeds_by_tid = mbeds.list_mbeds_by_targetid()
                if target_id in mbeds_by_tid:
                    if 'mount_point' in mbeds_by_tid[target_id]:
                        new_serial_port = mbeds_by_tid[target_id]['serial_port']
                        break
                sleep(to)
```